index.js

```diff
@@ -16,3 +17,228 @@ mobileLinks.forEach((link) => {
+
+projectsDataSource.forEach((project, index) => {
+  result
+  += `<div class="card grid__item mb-5">
+  <div class="inner-card d-flex p-4" id=${`card-${index}`}>
+  <div class="portfolio-image-container">
+  <img id="portfolio-img" src="${project.img}" alt="Project Snapshot" />
+  </div>
+  <div class="desc">
+  <h2 class="card-title">${project.title}</h2>
+  <p>${project.desc}</p>
+  <div class="list-inline p-1 d-flex">
+  ${project.techs.map((tech) => `<span class="badge p-2 text-primary mb-5 fw-normal">${tech}</span>`).join('')}
+  </div>
+  <button type="button" id="${index}" class="btn btn-outline-primary align-self-start see-project-btn"
+  onclick="projectClick(event)">See Project</button>
+  </div>
+  </div>
+  </div>`;
+});
+
+document.getElementById('Portfolio').innerHTML = result;
+
+const projects = [];
+
+const projectOne = {
+  id: '0',
+  img: './images/Snapshoot-Portfolio.svg',
+  name: 'Tonic',
+  tech: ['Codekit', 'GitHub', 'JavaScript'],
+  description: 'A daily selection of privately personalized reads; no accounts or sign-ups required.',
+  languages: ['HTML', 'CSS', 'JavaScript'],
+  demo: 'abc',
+  source: 'def',
+};
+
+projects.push(projectOne);
+
+const projectTwo = {
+  id: '1',
+  img: './images/Snapshoot-Portfolio2.svg',
+  name: 'Multi-Post Stories',
+  tech: ['Codekit', 'GitHub', 'JavaScript'],
+  description:
+    'Experimental content creation feature that allows users to add to an existing story over the course of a day without spamming their friends.',
+  languages: ['HTML', 'CSS', 'JavaScript'],
+  demo: 'abc',
+  source: 'def',
+
+};
+
+projects.push(projectTwo);
+
+const projectThree = {
+  id: '2',
+  img: './images/Snapshoot-Portfolio3.svg',
+  name: 'Facebook 360',
+  tech: ['Codekit', 'GitHub', 'JavaScript'],
+  description:
+    'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry`s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry`s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the relea',
+  languages: ['HTML', 'CSS', 'JavaScript'],
+  demo: 'abc',
+  source: 'def',
+};
+
+projects.push(projectThree);
+
+const projectFour = {
+  id: '3',
+  img: './images/Snapshoot-Portfolio4.svg',
+  name: 'Uber Navigation',
+  tech: ['Codekit', 'GitHub', 'JavaScript'],
+  description:
+    'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+  languages: ['HTML', 'CSS', 'JavaScript'],
+  demo: 'abc',
+  source: 'def',
+};
+
+projects.push(projectFour);
+/* eslint-disable */
+function projectClick(event) 
+{
+  const clickedId = event.target.id;
+  header.classList.toggle('header-toggler');
+
+  for (let i = 0; i < projects.length; i + 1) {
+    if (projects[i].id === clickedId) {
```

Comment:
I am currently not able to open up the project modal. An attempt to view a project causes an "Out of Memory" error as shown below:

This is because you're missing assigning a new incremental value to the i variable of the loop.

A better way to define the loop would have been something as this:
```javascript
   for (let i = 0; i < projects.length; i = i + 1) { // do }
```
You also have other loops with similar error.

Meanwhile, as a better practice, I suggest that you use the forEach statement for more readability.

![image](https://user-images.githubusercontent.com/20554583/232053247-2665f210-c41a-44e9-bf19-cd8680caa3f3.png)